### PR TITLE
Update Cloud docs for local runs and plan updates

### PR DESCRIPTION
**Pull request by d3rp3tt3**
## Note
This is a new PR to replace 6633. Something got off in the commit history for that PR after a rebase. It's easier to start afresh in a new PR. Original PR: https://github.com/dagger/dagger/pull/6633

## Overview
This PR includes updates to the Cloud docs
- [x] Clarify config and UX for local and CI
- [x] Update cache instructions to include config for pushing the volumes to distributed caching
- [x] Update getting started for UI changes, as needed
- [x] Update reference for UI changes, as needed
- [x] Update all Cloud docs for new plans

## Status
* Getting started guide: Ready for review
* Reference: In progress

We will also need to copy these docs into the Zenith/latest later. That will be in a separate PR.

### CODE REVIEW

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:19]:
```suggestion
- [Synchronize cache volumes with the Dagger Cloud experimental distributed cache (available on the Team plan)](#step-4-use-cache-volumes-with-the-experimental-dagger-cloud-cache)
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:49]: I don't think `*Team*` actually makes it bold since in the page you can barely notice the difference :thinking: 
image

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:55]: nit: Your dashboard will initially be blank. Do we need to be redundant on that? The user will see a similar message after landing to that page. Might not be necessary to re-iterate that here as well.

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:94]:
```suggestion
- For *Team* plan subscribers with ephemeral CI runners only: Add a wait step (example below) for the Docker configuration to allow the Dagger Engine to push cache volumes to the experimental Dagger Cloud distributed cache. Subscribers on the *Individual* plan do not have access to the experimental distributed cache and do not need to add this step.
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:99]: Unrelated to this PR, this danger note is a bit confusing. Shall we revisit this one @vikram-dagger ?

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:240]: missing screenshot. Just highlighting so we don't miss this.

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:244]: missing screenshot. Just highlighting so we don't miss this.

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:278]: nit: highlight experimental?

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (marcosnils) [inline docs/current_docs/cloud/572923-get-started.mdx:289]: I think we should revisit this paragraph in general as it puts too much emphasis on cache volumes vs caching operations. Let's put that in the queue @vikram-dagger

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (d3rp3tt3) [reply docs/current_docs/cloud/572923-get-started.mdx:47]: Thanks! I'll change it to: "Review the Dagger Cloud subscription plans and select one to move forward." Or an equivalent.

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (d3rp3tt3) [reply docs/current_docs/cloud/572923-get-started.mdx:49]: Yeah, that's making them italicized. I'm updating them to **text** in a commit coming in soon.

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (d3rp3tt3) [reply docs/current_docs/cloud/572923-get-started.mdx]: Updated in the 4 instances where this is mentioned

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (d3rp3tt3) [reply docs/current_docs/cloud/572923-get-started.mdx]: Fixed.

**Comment by d3rp3tt3** [comment]
I've added the Reference updates @marcosnils @slumbering @vikram-dagger

**Comment by d3rp3tt3** [comment]
DO NOT MERGE THIS PR UNTIL GO-LIVE on MONDAY - talk to Jenn first

File: docs/current_docs/cloud/reference/741031-user-interface.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/reference/741031-user-interface.mdx:17]: I'm not sure this statement needs to be in the docs, as there isn't a date on the document so it's hard to know when this experience will be available.

File: docs/current_docs/cloud/reference/741031-user-interface.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/reference/741031-user-interface.mdx:54]:
```suggestion
### Run Details page: Step viewer
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:19]: This list item doesn't match the heading title for the step.

Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:19]:
```suggestion
- [Use cache volumes with the experimental Dagger Cloud cache](#step-4-use-cache-volumes-with-the-experimental-dagger-cloud-cache)
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:70]:
```suggestion
  If you regenerate your token, you must replace the token wherever you've referenced it. To reduce operational interruptions, only regenerate your token if it has leaked. 
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:81]:
```suggestion
Dagger Cloud uses metadata from your version control system (VCS), such as Git, to populate some information in the user interface. Therefore, you should run your Dagger pipelines within a VCS context, such as a local directory linked to a cloned Git repository.
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [reply docs/current_docs/cloud/572923-get-started.mdx:99]: AFAIR we added it because it was a point of concern for AWS users.

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:94]: Note that this step is missing for Jenkins and Argo tabs. Do we need to add it, or is its absence intentional?

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:231]:
```suggestion
Here's an example running a Dagger Node.js pipeline named `index.mjs` locally after committing a change with Git.
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:238]:
```suggestion
Because we ran this in a Git context, the Dagger Cloud UI shows the VCS Git metadata associated with the commit. 
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:242]:
```suggestion
If you run outside of a Git context, Dagger Cloud will show the same detailed steps. However, the top-level metadata will not be populated. 
```

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [inline docs/current_docs/cloud/572923-get-started.mdx:246]: This is an odd place for this note. Perhaps move it to the end of the section about generating a token, or in the reference page?

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (vikram-dagger) [reply docs/current_docs/cloud/572923-get-started.mdx:289]: I didn't understand this @marcosnils . What change are you suggesting? That we should not mention cache volumes here and only focus on the experimental cache?

**Review by slumbering — APPROVED** [review]
:shipit:

File: docs/current_docs/cloud/572923-get-started.mdx
Comment (d3rp3tt3) [reply docs/current_docs/cloud/572923-get-started.mdx:246]: We're merging this ASAP. Let's pull any feedback into a new issue.

**Pull request merged**
